public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -217,26 +239,43 @@ function getElements(state: DashboardSceneState) {
           spec: {
             pluginVersion: vizPanel.state.pluginVersion ?? '',
             options: vizPanel.state.options,
-            fieldConfig: (vizPanel.state.fieldConfig as FieldConfigSource) ?? defaultFieldConfigSource(),
+            fieldConfig: vizFieldConfig ?? defaultFieldConfigSource(),
           },
         },
       },
     };
-    acc.push(elementSpec);
-    return acc;
-  }, []);
+
+    return elementSpec;
+  });
+
   // create elements
 
   const elements = createElements(panelsArray);
   return elements;
 }
 
 function getPanelLinks(panel: VizPanel): DashboardLink[] {
```

Comment:
This is wrongly typed in the v2 schema ATM, panel links should be DataLink, not dashboard link. With that I think the whole as DashboardLink is unnecessart.